shm/include/ship.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "cargo.hpp"
+
+class Ship {
+public:
+    Ship();
+    Ship(int capacity, int maxCrew, int speed, const std::string& name, size_t id);
+    Ship(int maxCrew, int speed, size_t id);
+
+    void setName(const std::string&);
+
+    Ship& operator-=(size_t);
+    Ship& operator+=(size_t num);
```

Comment:
One of operators has a name of variable the other hasn't. I think it's oversight.